subsys/logging/log_core.c

```diff
@@ -352,6 +359,43 @@ void log_n(const char *str,
 	}
 }
 
+int log_format_set(char *backend_name, uint32_t log_type)
+{
+	const struct log_backend *backend = log_backend_get_by_name(backend_name);
+
+	if (log_type >= ARRAY_SIZE(log_format_table)) {
+		return -EINVAL;
+	}
+
+	if (backend != NULL) {
+		if (backend->api->format_set == NULL) {
+			return -ENOTSUP;
+		} else {
+			return backend->api->format_set(backend, log_type);
+		}
+	} else {
+		return -EINVAL;
+	}
```

Comment:
Simplify. If we follow a pattern of bailing out early, we can reduce indentation (and this helps keep it easier to read).

if (backend == NULL) {
    return -EINVAL;
}

if (backend->api->format_set == NULL) {
    return -ENOTSUP;
}

return backend->api->format_set(backend, log_type);

Reply:
Made the changes to simplify this.